tokio-stream/src/wrappers/broadcast.rs

```diff
@@ -0,0 +1,61 @@
+use crate::Stream;
+use async_stream::stream;
+use std::pin::Pin;
+use tokio::sync::broadcast::error::RecvError;
+use tokio::sync::broadcast::Receiver;
+
+use std::fmt;
+use std::task::{Context, Poll};
+
+/// A wrapper around [`Receiver`] that implements [`Stream`].
+///
+/// [`Receiver`]: struct@tokio::sync::broadcast::Receiver
+/// [`Stream`]: trait@crate::Stream
+pub struct BroadcastStream<T: Clone> {
+    inner: Pin<Box<dyn Stream<Item = Result<T, BroadcastStreamRecvError>> + Send + Sync>>,
+}
+
+/// An error returned from the inner stream of a [`BroadcastStream`].
+#[derive(Debug, PartialEq)]
+pub enum BroadcastStreamRecvError {
+    /// The receiver lagged too far behind. Attempting to receive again will
+    /// return the oldest message still retained by the channel.
+    ///
+    /// Includes the number of skipped messages.
+    Lagged(u64),
+}
+
+impl<T: Clone + Unpin + 'static + Send + Sync> BroadcastStream<T> {
+    /// Create a new `BroadcastStream`.
+    pub fn new(mut rx: Receiver<T>) -> Self {
+        let stream = stream! {
+            loop {
+                match rx.recv().await {
+                Ok(item) => yield Ok(item),
+                Err(err) =>
```

Comment:
rustfmt doesn't work inside macros, but it would be nice to indent this more nicely.